plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -201,5 +202,4 @@ instance ExMemoryUsage Char where
 instance ExMemoryUsage Bool where
   memoryUsage _ = 1
 
-instance ExMemoryUsage String where
-  memoryUsage string = ExMemory $ fromIntegral $ (sum $ fmap sizeOf string) `div` 8
+deriving via GenericExMemoryUsage [a] instance ExMemoryUsage a => ExMemoryUsage [a]
```

Comment:
This worries me.  If I understand correctly (and I may well have got it completely wrong), this will get the memory usage of a built-in list by traversing the entire list and adding up the memory usage of its members, along with some contribution from the constructors.  I think there are two things wrong with this. (A) if we're applying a built-in function to a list then we'll have to get the memory usage of the list just before applying the function, and this will preform a potentially expensive traversal.  (B) I think it's the wrong metric: we probably want the length of the list here. If we want to eg append one list to another then the cost will depend on the length of the lists (in fact, just the first list) and we don't care whether the entries are small booleans or massive bytestrings, but the sizes of those will be included in the generic memory usage. 

See also the main comment on this PR.

Reply:
That's a good point, but we were using `fmap` before this PR as well, so the PR doesn't change much, so let's ignore the problem for now.